src/ol_infrastructure/components/aws/s3_cloudfront_site.py

```diff
@@ -0,0 +1,118 @@
+"""Module for creating and managing static websites hosted in S3 and delivered through Cloudfront."""
+from enum import Enum
+from typing import List, Text, Optional
+
+from pydantic import validator
+from pulumi import ComponentResource, ResourceOptions
+from pulumi_aws import acm, cloudfront, s3
+
+from ol_infrastructure.lib.ol_types import AWSBase
+
+
+class CloudfrontPriceClass(str, Enum):  # noqa: WPS600
+    """Valid price classes for CloudFront to control tradeoffs of price vs. latency for global visitors."""
+
+    # For more details on price class refer to below link and search for PriceClass
+    # https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_DistributionConfig.html
+    # NB: POP == Points Of Presence which is where CDN edge servers are located
+    us_eu = 'PriceClass_100'  # POPs in US, Canada, Europe, and Israel
+    exclude_au_sa = 'PriceClass_200'  # POPs in all supported geos except South America and Australia
+    all_geos = 'PriceClass_All'  # POPs in all supoprted geos
+
+
+class S3ServerlessSiteConfig(AWSBase):
+    """Configuration object for customizing a static site hosted with S3 and Cloudfront."""
+
+    site_name: Text
+    domains: List[Text]
+    bucket_name: Text
+    site_index: Text = 'index.html'
+    cloudfront_price_class: CloudfrontPriceClass = CloudfrontPriceClass.us_eu
```

Comment:
Should that price class be hardcoded or is there a way to override it?

Reply:
It's not hardcoded, that's just a default. This acts as a data class which gets instantiated for the specific instances.